schemas/dbt_yml_files.json

```diff
@@ -165,32 +165,27 @@
         "type": "array",
         "items": {
           "type": "object",
-          "required": ["name", "type", "sql", "timestamp", "time_grains"],
```

Comment:
This is technically redundant, but it enables slightly smarter autocomplete suggestions - otherwise the suggestions can’t come through until its calculation_method is set (and maybe not even then?).

Reply:
would we want these to be autofilled if several will be deleted in the case of a derived metric? would we prefer to default to "regular mode" and allow users to pivot to derived if they want?

Reply:
Tricky question 🤔 My original reply was going to be: 
>I think assume regular mode and let people remove them if they're unnecessary/irrelevant. If you do a derived metric, `model` is not only not required but actually invalid, right? Do we have any way of marking that as invalid?

But thinking about it more, maybe it makes more sense to just do the common ones and then once they set the `type` to somethig other than `expression` the validator will complain that they haven't got `model` provided. And I think that makes more sense than prefilling something and confusing people about what model means in an expression metric

Reply:
i went through this process when developing locally! i think best to autofill as far as we can with common ones, then let the validation remind folks about `model` if and when they don't define the `calculation_method: derived`